substratevm/src/com.oracle.svm.jfr/src/com/oracle/svm/jfr/JfrManager.java

```diff
@@ -217,6 +204,43 @@ private static Long parseDuration(Map<JfrStartArgument, String> args, JfrStartAr
         return null;
     }
 
+    private static Long parseMaxSize(Map<JfrStartArgument, String> args, JfrStartArgument key) {
+        final String value = args.get(key);
+        if (value != null) {
+            try {
+                int idx = indexOfFirstNonDigitCharacter(value);
+                long number;
+                try {
+                    number = Long.parseLong(value.substring(0, idx));
+                } catch (NumberFormatException e) {
+                    throw new IllegalArgumentException("Expected a number.");
+                }
+
+                // Missing unit, number is plain bytes
+                if (idx == value.length()) {
+                    return number;
```

Comment:
@jiekang The default case is supported here, i.e. no unit means it's just bytes.

See:

```
$ ./main -XX:+FlightRecorder -XX:StartFlightRecording=maxsize=10000000,filename=flight-native.jfr -XX:FlightRecorderLogging=jfr
[info][jfr] Flight Recorder initialized
[info][jfr] Created repository /tmp/2021_07_30_15_20_59_298189
[info][jfr       ] Started recording "1" (1) {maxsize=9.5MB, dumponexit=true, filename=/tmp/rep/flight-native.jfr}
Meh.
```